pkg/kubectl/resize.go

```diff
@@ -70,11 +93,28 @@ type ReplicationControllerResizer struct {
 	client.Interface
 }
 
+// ResizeCondition is a closure around Resize that facilitates retries via util.wait
+func ResizeCondition(r Resizer, precondition *ResizePrecondition, namespace, name string, count uint) wait.ConditionFunc {
+	return func() (bool, error) {
+		_, err := r.Resize(namespace, name, precondition, count)
+		switch err.(type) {
+		case nil:
+			return true, nil
+		case ControllerResizeError:
+			e := err.(ControllerResizeError)
+			if e.FailureType == UpdateFailure {
+				return false, nil
```

Comment:
I thought about this but decided to punt because there was no nice way to classify the errors coming out of client/request